### PR TITLE
feat: Attestations

### LINKED ISSUE #681: SAR: Attestations
**Extended Description**
-As an ASAP developer, in order to assist FedRAMP reviewers, I want to ensure that a result has appropriate attestation information.

**Acceptance Criteria**
- [x] All Schematron assertion messages are declarative statements which affirm the positive test outcome.
- [x] All Schematron assertion diagnostic messages are declarative statements which explain the negative test outcome.
- [x] The Schematron code has no assertion failures when validated using `src/validations/styleguides/sch.sch` using the `basic` phase.
- [x] XSpec unit tests for positive and negative Schematron assertion outcomes accompany all Schematron assertions (where feasible).
- [x] result elements have attestation/part[@name='authorization-statements']/prop with attributes @ns='https://fedramp.gov/ns/oscal' and @name='recommend-authorization'
- [x] When the above, check the @value value.  When any value other than 'yes' the first paragraph of the first part child 'authorization-statement' must conform to the following:
"A total of [# of risks] system risks were identified for [system name], including [#high] High risks, [#moderate] Moderate risks, [#low] Low risks, and [#operationally-required] of operationally required risks."
- [x] Each risk element may have a prop with a @name='priority' and @ns='https://fedramp.gov/ns/oscal'.  If it does then the value must be a number that is unique among the risk elements.  

**Definition of Done**
 - [x] Acceptance criteria met
 - [ ] Unit test coverage of our code > 95%
 - [ ] Automated code quality checks passed
 - [ ] Security reviewed and reported
 - [ ] Reviewed against plain language guidelines
 - [ ] Code must be self-documenting
 - [ ] No local tech debt
 - [ ] Load/performance tests passed – needs to be created/automated
 - [ ] Documentation updated
 - [ ] Architectural Decision Record completed as necessary for significant design choices
 - [ ] PR reviewed & approved
 - [ ] Source code merged

### PR DESCRIPTION
**Pull request by markXLIX**
Closes #681 
* result elements have attestation/part[@name='authorization-statements']/prop with attributes @ns='https://fedramp.gov/ns/oscal' and @name='recommend-authorization'
* When the above, check the @value value. When any value other than 'yes' the first paragraph of the first part child 'authorization-statement' must conform to the following:
"A total of [# of risks] system risks were identified for [system name], including [#high] High risks, [#moderate] Moderate risks, [#low] Low risks, and [#operationally-required] of operationally required risks."
* Each risk element may have a prop with a @name='priority' and @ns='https://fedramp.gov/ns/oscal'. If it does then the value must be a number that is unique among the risk elements.

### CODE REVIEW

File: src/validations/rules/sar.sch [hunk 1]
```diff
@@ -176,6 +176,65 @@
...
+
+        <sch:rule
+            context="oscal:result">
+            <sch:assert
+                diagnostics="has-attestation-diagnostic"
+                fedramp:specific="true"
+                id="has-attestation"
+                role="error"
+                test="oscal:attestation[oscal:part[@name = 'authorization-statements']/oscal:prop[@ns = 'https://fedramp.gov/ns/oscal' and @name = 'recommend-authorization']]">
+                There must exist an attestation with a part containing a property with a name of 'recommend-authorization'.</sch:assert>
```
Comment (GaryGapinski) [inline src/validations/rules/sar.sch:193]: Remove " a part containing".

**Review by GaryGapinski — CHANGES_REQUESTED** [review]

File: src/validations/rules/sar.sch [hunk 2]
```diff
@@ -342,6 +338,62 @@
...
+            <sch:assert
+                diagnostics="has-attestation-value-no-diagnostic"
+                fedramp:specific="true"
+                id="has-attestation-value-no"
+                role="error"
+                see="Guide to OSCAL-based FedRAMP Security Assessment Reports - Section 4.12"
+                test="
+                    if (oscal:prop/@value ne 'yes')
+                    then
+                        if (matches(normalize-space(oscal:part[1]/oscal:p[1]), 'A total of .* system risk(s?) were identified for .*, including .* High risk(s?), .* Moderate risk(s?), .* Low risk(s?), and .* of operationally required risk(s?).'))
```
Comment (GaryGapinski) [inline src/validations/rules/sar.sch:389]: `.*` has cardinality {0,}; is that correct (i.e., nothing but two spaces in those locations is acceptable).
If these are going to be numbers as opposed to anything, try using `\d+`.

**Review by GaryGapinski — CHANGES_REQUESTED** [review]
One question.

File: src/validations/rules/sar.sch (hunk 2 above)
Comment (markXLIX) [reply src/validations/rules/sar.sch:389]: I didn't feel confident from the Guide narrative that the value would always be a digit.  I thought of [a-zA-Z0-9]* but also wasn't sure if that was overkill.

I'll adjust to add the alpha and digit instead.

Comment (markXLIX) [reply src/validations/rules/sar.sch]: 2nd thought.  Went with \w+ to accommodate the digits and alphas as well as state that there couldn't be empty space.

**Review by GaryGapinski — APPROVED** [review]

**Pull request merged**